pkg/distributor/ha_tracker.go

```diff
@@ -476,19 +483,27 @@ func (h *haTracker) updateKVStore(ctx context.Context, userID, cluster, replica
 		var ok bool
 		if desc, ok = in.(*ReplicaDesc); ok && desc.DeletedAt == 0 {
 			// If the entry in KVStore is up-to-date, just stop the loop.
-			if h.withinUpdateTimeout(now, desc.ReceivedAt) ||
-				// If our replica is different, wait until the failover time.
-				desc.Replica != replica && now.Sub(timestamp.Time(desc.ReceivedAt)) < h.cfg.FailoverTimeout {
+			if h.withinUpdateTimeout(now, desc.ReceivedAt) {
+				return nil, false, nil
+			}
+		}
+		// If our replica is different, wait until the failover time
+		if desc != nil && desc.Replica != replica {
+			if now.Sub(timestamp.Time(desc.ReceivedAt)) < h.cfg.FailoverTimeout {
+				level.Info(h.logger).Log("msg", "replica differs, but it's too early to failover", "user", userID, "cluster", cluster, "replica", replica, "elected", desc.Replica, "received_at", timestamp.Time(desc.ReceivedAt))
 				return nil, false, nil
 			}
+			level.Info(h.logger).Log("msg", "replica differs, attempting to update kv", "user", userID, "cluster", cluster, "replica", replica, "elected", desc.Replica, "received_at", timestamp.Time(desc.ReceivedAt))
 		}
 
 		// Attempt to update KVStore to our timestamp and replica.
 		desc = &ReplicaDesc{
 			Replica:    replica,
 			ReceivedAt: timestamp.FromTime(now),
 			DeletedAt:  0,
+			ElectedAt:  timestamp.FromTime(now),
 		}
+		h.lastElectionTimestamp.WithLabelValues(userID, cluster).Set(float64(desc.ReceivedAt / 1000))
```

Comment:
Similar to the comment above, this will be updated when we just update the received at. Plus it will differ from distributor to distributor. 

I'd say we should update this outside of the CAS, only if CAS succeeded and we update the replica, not just the timestamp. However, that would mean that if all distributors restart and no reelection happened, no metric is exposed, so maybe we could expose this metric from the `updateCache` code?

Scope creep: in order to avoid adding cardinality (distributors times users), we could expose this metrics from the overrides-exporter (it would have to start an ha tracker).